sdk/typescript/src/utils/gas-estimation.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { UnserializedSignableTransaction } from '../signers/txn-data-serializers/txn-data-serializer';
+import { Coin, SuiMoveObject } from '../types';
+
+export type TxKind = UnserializedSignableTransaction['kind'];
+type BudgetMap = typeof DEFAULT_GAS_BUDGET_PER_TX_TYPE;
+export type GasBudgetGuessParams<txKind extends TxKind> =
+  txKind extends keyof BudgetMap
+    ? BudgetMap[txKind] extends (...args: any[]) => any
+      ? Parameters<BudgetMap[txKind]> extends [boolean, ...infer Params]
+        ? Params
+        : []
+      : []
+    : [];
+
+export const MIN_GAS_BUDGET = 10;
+export const MAX_GAS_BUDGET = 1_000_000;
+
+const PAY_GAS_FEE_PER_COIN = 150;
+
+/**
+ * Make a guess for the gas budget of a pay tx
+ * @param coins All the coins of the type to send
+ * @param amount The amount to send
+ * @returns The gasBudget guess
+ */
+function computeTransferTxGasBudget(
```

Comment:
I'm confused why this can't be done with dry run.

Reply:
Same as the above, and also in case of dry-run fails we use this estimation